packages/core/src/Serialization/TransactionBody/TransactionBody.ts

```diff
@@ -132,19 +132,24 @@ export class TransactionBody {
 
     if (this.#withdrawals && this.#withdrawals.size > 0) {
       writer.writeInt(5n);
-
-      const sortedCanonically = new Map([...this.#withdrawals].sort((a, b) => (a > b ? 1 : -1)));
-
-      writer.writeStartMap(sortedCanonically.size);
-
-      for (const [key, value] of sortedCanonically) {
+      // Create a new map with address bytes to avoid logic duplication and address checks
+      const withdrawalsWithAddressBytes = new Map();
+      for (const [key, value] of this.#withdrawals) {
         const rewardAddress = Cardano.RewardAddress.fromAddress(Cardano.Address.fromBech32(key));
-
         if (!rewardAddress) {
           throw new SerializationError(SerializationFailure.InvalidAddress, `Invalid withdrawal address: ${key}`);
         }
+        const rewardAddressBytes = rewardAddress.toAddress().toBytes();
+        withdrawalsWithAddressBytes.set(rewardAddressBytes, value);
+      }
+
+      // Sort withdrawals by address bytes, canonically
+      const sortedCanonically = new Map([...withdrawalsWithAddressBytes].sort((a, b) => (a > b ? 1 : -1)));
```

Comment:
A map here looks odd since we're looking for ordered collection.
```suggestion
      const sortedCanonically = [...withdrawalsWithAddressBytes].sort((a, b) => (a > b ? 1 : -1));
```

Reply:
this will create an array and we need a Map. I reused the existing logic but transformed addresses to address bytes
https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/core/src/Serialization/TransactionBody/TransactionBody.ts#L136

Reply:
Why do we need a map? for-of where it's used should work with array just as well :thinking:

Reply:
aaaah, you are right, please ignore my last comment. Sorry